system/src/system_network_cellular.h

```diff
@@ -80,10 +85,12 @@ class CellularNetworkInterface : public ManagedIPNetworkInterface<CellularConfig
     int on_now() override {
         cellular_result_t ret = cellular_on(nullptr);
         if (ret != 0) {
+            resume_if_needed();
             return ret;
         }
         ret = cellular_init(nullptr);
         if (ret != 0) {
+            resume_if_needed();
```

Comment:
Having `resume_if_needed()` here might still have the modem go thru the `cellular_init()` process, correct? in which case it will still take time. I too believe that the `resume_if_needed()` should be moved right to the beginning of `on_now()`. Please correct me otherwise.